src/openforms/submissions/api/viewsets.py

```diff
@@ -108,60 +102,64 @@ def _complete(self, request, *args, **kwargs):
-        no longer possible to change or read the submission data (including individual
-        steps).
+        Note that the processing of the submission runs in the background, and you
+        should periodically check the submission status endpoint to check the state.
+        Background processing makes sure that:
+
+        * potential appointments are registered
+        * a report PDF is generated
+        * the submission is persisted to the configured backend
+        * payment is initiated if relevant
+
+        Once a submission is done processing, it's scheduled to be removed from the
+        session. This means it's no longer possible to change or read the submission
+        data (including individual steps).
 
-        The submission is persisted to the configured backend.
+        TODO: give some leeway? add endpoint to trigger cleanup immediately?
+        TODO: emit a status URL with time-based token (Salted HMAC, from token generator)
+        that the frontend can poll without needing the submission in the session.
+        TODO: if appointment registration fails, the submission ID must be added back
+        to the session so the user can return to the first step(s) in the UI.
         """
         submission = self.get_object()
         validate_submission_completion(submission, request=request)
-        submission.completed_on = timezone.now()
 
+        submission.completed_on = timezone.now()
         submission.save()
+
+        # TODO: implement in celery tasks in cleanup (see ./tasks/__init__.py)
         remove_submission_from_session(submission, self.request.session)
         remove_submission_uploads_from_session(submission, self.request.session)
 
-        submission_report = SubmissionReport.objects.create(
-            title=_("%(title)s: Submission report") % {"title": submission.form.name},
-            submission=submission,
-        )
-
-        def on_submission_commit():
-            # The submission report needs to already have been generated before it can be attached to the zaak
-            # that is created in the registration
-            chain = generate_submission_report.si(
-                submission_report.id
-            ) | register_submission.si(submission.id)
-            chain.delay()
-
-            # this can run any time because they have been claimed earlier
-            cleanup_temporary_files_for.delay(submission.id)
-
-        transaction.on_commit(on_submission_commit)
-
-        book_appointment_for_submission(submission)
+        # after committing the database transaction where the submissions completion is
+        # stored, start processing the completion.
+        transaction.on_commit(lambda: on_completion(submission.id))
 
-        if hasattr(submission.form, "confirmation_email_template"):
-            transaction.on_commit(lambda: send_confirmation_email(submission))
+        return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
In a separate PR, I'll look at the status endpoint and ensure that the URL is included here - this will probably be token-based to not need the submission ID in the session, yet return a URL that will expire after a set time and/or internal state change.